torch/csrc/jit/mobile/import.cpp

```diff
@@ -288,6 +276,71 @@ TypePtr BytecodeDeserializer::resolveTypeName(const c10::QualifiedName& qn) {
   return resolveTypeNameMobile(qn, compilation_unit_);
 }
 
+// It requires compilation_unit_ when parsing function schema. Keep it in
+// BytecodeDeserializer. It may be refacotred later to make it independent
+// of the specific BytecodeDeserializer, like parsing other tables
+void BytecodeDeserializer::parseFunctionSchema(
+    const std::string& function_name,
+    const at::optional<IValue>& schemaTable,
+    const int64_t& model_version,
+    mobile::Function* function) {
+  // function schema
+  if (schemaTable) { // (schema is optional for back compat)
+    auto parseArgList = [this](const std::vector<IValue>& argTables) {
+      std::vector<c10::Argument> args;
+      for (auto&& argTable : argTables) {
+        auto name = expect_field(argTable, "name", BYTECODE_INDEX_ARGUMENT_NAME)
+                        .toStringRef();
+        const auto& type = resolveTypeName(
+            (expect_field(argTable, "type", BYTECODE_INDEX_ARGUMENT_TYPE))
+                .toStringRef());
+        auto default_value = expect_field(
+                                 argTable,
+                                 "default_value",
+                                 BYTECODE_INDEX_ARGUMENT_DEFAULT_VALUE)
+                                 .toIValue();
+        auto arg = c10::Argument(name, type, c10::nullopt /*N*/, default_value);
+        args.emplace_back(std::move(arg));
+      }
+      return args;
+    };
+    const auto& arg_list =
+        expect_field(*schemaTable, "arguments", BYTECODE_INDEX_SCHEMA_ARGUMENTS)
+            .toTuple()
+            ->elements();
+    const auto& ret_list =
+        expect_field(*schemaTable, "returns", BYTECODE_INDEX_SCHEMA_RETURNS)
+            .toTuple()
+            ->elements();
+    c10::FunctionSchema schema(
+        function_name,
+        "" /*overload_name*/,
+        parseArgList(arg_list),
+        parseArgList(ret_list),
+        false /*is_varargs*/,
+        false /*is_varret*/);
+    function->setSchema(std::move(schema));
+  }
+}
+
+void parseOperators(
```

Comment:
Looks like Instructions/Constants/Types are moved to the new source file, while Operators/FunctionSchema are kept here. Are we trying to separate the parsing functions into two set, one set of core functions, one set of customizable functions? How do we decide which should be in which set?

Reply:
nit: why parseOperators are not in BytecodeDeserializer?

Reply:
FunctionSchema is optional. It's majorly used to handle default arguments in Module's methods. I don't think it's needed to a minimum env like embedded but I could be wrong. 

parseOperators() is now tightly coupled with PyTorch's operator resolver (for example, `c10::Dispatcher::singleton().findSchema(opname)`. I'm not sure if the same "operator" resolution (finder) would be used for all backends. If so, we could pull it from the original loader. One option I could think of is to pass different operator resolution functions with the same signature `*function findOperator(opname)`. Another option is to reuse the PyTorch's operator resolution, but decouple it from the bigger ATen/Core.

Reply:
@raziel I agree that "pass different operator resolution functions with the same signature `*function findOperator(opname)`" would be a better option. 

For "FunctionSchema" here, it's the schema for a Module's function (like forward()), not the schema for an operator. We could support this, but maybe in a layered way? The reason is that the schema parser would take some binary size.